File: sanket-dalvi/Day3/countSquares.py
class Solution:
    def countSquares(self, N):
        def sSearch(l, h):
            if l < h:
                mid = (l+h)//2
                if mid**2 == N:
                    return mid
                elif mid**2 < N:
                    return sSearch(mid+1, h)
                else:
                    return sSearch(l, mid)
            else:
                return l
        return sSearch(1, N)-1

# Test
print(Solution().countSquares(10))

File: Adhiti_Shaili/Day 3/Count Squares.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    int countSquares(int N) {
        int n=pow(N,0.5);
        int count=0;
        for(int i=1;i<=n;i++)
        {
            if(i*i<N)
            {
                count++;
            }
        }
        return count;
    }
};

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        
        cin>>N;

        Solution ob;
        cout << ob.countSquares(N) << endl;
    }
    return 0;
} 